### PR TITLE
Logistic Regression CI fix

### PR DESCRIPTION
**Pull request by md-shafiul-alam**
### Description

test_csr fails for logistic regression in Nightly, increasing tolerance.

---

Checklist to comply with before moving PR from draft:

**PR completeness and readability**

- [x] I have reviewed my changes thoroughly before submitting this pull request.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have updated the documentation to reflect the changes or created a separate PR with update and provided its number in the description, if necessary.
- [x] Git commit message contains an appropriate signed-off-by string _(see [CONTRIBUTING.md](https://github.com/intel/scikit-learn-intelex/blob/main/CONTRIBUTING.md#pull-requests) for details)_.
- [x] I have added a respective label(s) to PR if I have a permission for that.  
- [x] I have resolved any merge conflicts that might occur with the base branch.

**Testing**

- [x] The unit tests pass successfully.
- [x] I have run it locally and tested the changes extensively.

**Performance**

- [x] I have measured performance for affected algorithms using [scikit-learn_bench](https://github.com/IntelPython/scikit-learn_bench) and provided at least summary table with measured data, if performance change is expected.
- [x] I have provided justification why performance has changed or why changes are not expected.

### CODE REVIEW

File: sklearnex/linear_model/tests/test_logreg.py [hunk 1]
```diff
@@ -129,5 +129,5 @@ def test_csr(queue, dtype, dims):
 
         assert_allclose(pred, pred_sp)
         assert_allclose(prob, prob_sp)
-        assert_allclose(model.coef_, model_sp.coef_)
-        assert_allclose(model.intercept_, model_sp.intercept_)
+        assert_allclose(model.coef_, model_sp.coef_, atol=1e-6)
```
Comment (avolkov-intel) [inline sklearnex/linear_model/tests/test_logreg.py:132]: you can increase tolerance up to 1e-5 to avoid sporadic failures

Comment (ethanglaser) [reply sklearnex/linear_model/tests/test_logreg.py:132]: Agreed, or even 1e-4. I see in the nightly that max relative difference was 7.3e-5

Comment (ethanglaser) [reply sklearnex/linear_model/tests/test_logreg.py:132]: I believe rtol is a better metric since its independent of data

Comment (md-shafiul-alam) [reply sklearnex/linear_model/tests/test_logreg.py]: done

**Review by icfaust — APPROVED** [review]
1e-4 relative tolerance is a bit eye opening (or ~7e-5). I am going to approve it with the understanding that a follow up ticket for an investigation is made.

**Comment by avolkov-intel** [comment]
@Mergifyio copy rls/2024.7.0-rls
